### PR TITLE
Integrate Firebase Analytics in the Javascript code.

### PR DESCRIPTION
**Pull request by Anubhople**
|PR checklist|Status| 
|-----|:------:| 
|Is issue created in GitHub?|[Yes](https://github.com/maqsoftware/GLEXP-Team-CCI-Hindi-Assets/issues/74)| 
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|NA| 
|Is code analysis task executed and issues resolved in the CPP/native code?|NA| 
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|NA| 
|Is code commenting done properly for the changes made?|Yes| 
|Is correct variable naming convention followed?|Yes| 
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|Yes| 
|Are the message strings used in the application taken from strings.xml file?|NA| 
|Are the permissions which are not required removed from AndroidManifest.java file?|NA| 
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA| 
|Are the generated files correctly configured in .gitignore?|NA| 
|Are the release signing configurations removed from the gradle files?|NA| 
|Is commented code removed from all files?|Yes| 
|Are updated images compressed (< 100 KB)?|NA| 
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

### CODE REVIEW

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js [hunk 1]
```diff
@@ -1,19 +1,55 @@
 // Main class for all android/ios app modifications (assuming cordova)
+var endTime, startTime;
+function Calculate_time() {
+
+    var url = window.location.pathname.split("/");
+    url.pop(); // index.html
+    var book = url.pop();
+    var subject = url.pop();
+    var Module_name = url.pop();
```
Comment (Shailin) [inline EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Use proper naming conventions for the variable name

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js [hunk 2]
```diff
@@ -1,19 +1,55 @@
...
+    }
+    var Module_Name = removeSpecialChars(Module_name);
+    var Level_Name = removeSpecialChars(subject);
+    var Lesson_Name = removeSpecialChars(book);
+
+    endTime = new Date();
+    var timeDiff = endTime - startTime; //in seconds
+    timeDiff /= 1000;
+    var Elapsed_Time = Math.round(timeDiff);
+    console.log(Elapsed_Time + " Elapsed_Time");
```
Comment (Shailin) [inline EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Remove logs if not required

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js (hunk 1 above)
Comment (Anubhople) [reply EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Changed variable names according to naming conventions.

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js (hunk 2 above)
Comment (Anubhople) [reply EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Removed.

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js [hunk 3]
```diff
@@ -1,19 +1,55 @@
...
+    endTime = new Date();
+    var timeDiff = endTime - startTime; //in seconds
+    timeDiff /= 1000;
+    var elapsedTime = Math.round(timeDiff);
+
+    document.addEventListener("deviceready", function () {
+        window.FirebasePlugin.setAnalyticsCollectionEnabled(true);
+        window.FirebasePlugin.setScreenName("Module_Play_Status");
+        // Log only the module name for the Epic Quest section
+        if (lessonName == "Epic    Quest") {
```
Comment (akashdabhi03) [inline EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Is the lesson name "Epic    Quest" along with these spaces?

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js [hunk 4]
```diff
@@ -1,4 +1,41 @@
...
+             var timeDiff = endTime - startTime; //in seconds
+             timeDiff /= 1000;
+             var Elapsed_Time = Math.round(timeDiff);
+             console.log(Elapsed_Time + " Elapsed_Time");
+
+             document.addEventListener("deviceready", function(){
+             window.FirebasePlugin.setAnalyticsCollectionEnabled(true);
+             window.FirebasePlugin.setScreenName("Module_Play_Status");
+
+            if(Lesson_Name=="Epic    Quest")
```
Comment (SoniaChhabria) [inline EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Use if(Lesson_Name.toLowerCase() === "epic quest")

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js [hunk 5]
```diff
@@ -1,4 +1,41 @@
...
+             var Module_Name= removeSpecialChars(Module_name);
+             var Level_Name= removeSpecialChars(subject);
+             var Lesson_Name= removeSpecialChars(book);
+
+             endTime = new Date();
+             var timeDiff = endTime - startTime; //in seconds
+             timeDiff /= 1000;
+             var Elapsed_Time = Math.round(timeDiff);
+             console.log(Elapsed_Time + " Elapsed_Time");
+
```
Comment (SoniaChhabria) [inline EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Remove console.log()

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js (hunk 3 above)
Comment (SoniaChhabria) [inline EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Use if(lessonName.Replace("   ","").toLowerCase() === "epicquest")

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js [hunk 6]
```diff
@@ -1,19 +1,55 @@
...
+    // remove the special characters found in the URL string
+    function removeSpecialChars(str) {
+        return str.replace(/%20/g, "    ");
+    }
+    var moduleName = removeSpecialChars(moduleName);
+    var levelName = removeSpecialChars(subject);
+    var lessonName = removeSpecialChars(book);
+
+    endTime = new Date();
+    var timeDiff = endTime - startTime; //in seconds
```
Comment (akashdabhi03) [inline EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js:20]: startTime is not calculated before the function is called. Are you getting correct value for timeDiff?

**Review by akashdabhi03 — CHANGES_REQUESTED** [review]

File: EnglishWebRoot/school/script/load.js [hunk 7]
```diff
@@ -17,11 +18,47 @@ window.addEventListener("popstate", function (e) {
...
+    endTime = new Date();
+    var timeDiff = endTime - startTime; //in seconds
+    timeDiff /= 1000;
+    var elapsedTime = Math.round(timeDiff);
+
+    document.addEventListener("deviceready", function () {
+        window.FirebasePlugin.setAnalyticsCollectionEnabled(true);
+        window.FirebasePlugin.setScreenName("Module_Play_Status");
+        // Log only the module name for the tutorial section
+        if (levelName == "tutorials") {
```
Comment (Shailin) [inline EnglishWebRoot/school/script/load.js]: levelName === "tutorials"

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js (hunk 4 above)
Comment (Anubhople) [reply EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Added this code.

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js (hunk 3 above)
Comment (Anubhople) [reply EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Added this code.

Comment (Anubhople) [reply EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js]: Removed the spaces.

File: EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js (hunk 6 above)
Comment (Anubhople) [reply EnglishWebRoot/school/pubbly_engine/latest/js/app/_AppModifications.js:20]: Yes its working fine.

File: EnglishWebRoot/school/script/load.js (hunk 7 above)
Comment (Anubhople) [reply EnglishWebRoot/school/script/load.js]: Modified the code.

**Review by Shailin — APPROVED** [review]

**Review by akashdabhi03 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

**Pull request merged**